datasets/imppres/imppres.py

```diff
@@ -0,0 +1,318 @@
+                    gen_kwargs={
+                        "filepath": os.path.join(data_dir, "possessed_definites_uniqueness.jsonl"),
+                        "split": "possessed_definites_uniqueness",
+                    },
+                ),
+                datasets.SplitGenerator(
+                    name="question_presupposition",
+                    # These kwargs will be passed to _generate_examples
+                    gen_kwargs={
+                        "filepath": os.path.join(data_dir, "question_presupposition.jsonl"),
+                        "split": "question_presupposition",
+                    },
+                ),
+            ]
+
+        else:
+            return [
+                datasets.SplitGenerator(
+                    name="connectives",
+                    # These kwargs will be passed to _generate_examples
+                    gen_kwargs={"filepath": os.path.join(data_dir, "connectives.jsonl"), "split": "connectives"},
+                ),
+                datasets.SplitGenerator(
+                    name="gradable_adjective",
+                    # These kwargs will be passed to _generate_examples
+                    gen_kwargs={
+                        "filepath": os.path.join(data_dir, "gradable_adjective.jsonl"),
+                        "split": "gradable_adjective",
+                    },
+                ),
+                datasets.SplitGenerator(
+                    name="gradable_verb",
+                    # These kwargs will be passed to _generate_examples
+                    gen_kwargs={"filepath": os.path.join(data_dir, "gradable_verb.jsonl"), "split": "gradable_verb"},
+                ),
+                datasets.SplitGenerator(
+                    name="modals",
+                    # These kwargs will be passed to _generate_examples
+                    gen_kwargs={"filepath": os.path.join(data_dir, "modals.jsonl"), "split": "modals"},
+                ),
+                datasets.SplitGenerator(
+                    name="numerals_10_100",
+                    # These kwargs will be passed to _generate_examples
+                    gen_kwargs={
+                        "filepath": os.path.join(data_dir, "numerals_10_100.jsonl"),
+                        "split": "numerals_10_100",
+                    },
+                ),
+                datasets.SplitGenerator(
+                    name="numerals_2_3",
+                    # These kwargs will be passed to _generate_examples
+                    gen_kwargs={"filepath": os.path.join(data_dir, "numerals_2_3.jsonl"), "split": "numerals_2_3"},
+                ),
+                datasets.SplitGenerator(
+                    name="quantifiers",
+                    # These kwargs will be passed to _generate_examples
+                    gen_kwargs={"filepath": os.path.join(data_dir, "quantifiers.jsonl"), "split": "quantifiers"},
+                ),
+            ]
```

Comment:
Could you use configurations instead of splits to differentiate between all the files ?

We use splits only for separations used during the evaluation of models (typically train/dev/test)

Reply:
I'm happy to do that, I just want to make sure that's the right action. The dataset is only used for testing already trained models. It has an `implicature` and `presupposition` configuration. Underneath each configuration it has the above sub-datasets. But all of these sub-datasets are only used for testing.

I'm happy to move them to configurations, but I would ask what would be the best way to arrange them? Something like 
```python
datasets.BuilderConfig(
            name="presupposition_change_of_state",
            version=VERSION,
            description="Presuppositions are facts that the speaker takes for granted when uttering a sentence.",
        ),
```
etc. or some other way?

Reply:
Yes sounds good.

You can have one configuration for each file type, and for each config only have one test set.

Reply:
Done!